Decisions/scs-XXXX-v1-requirements-for-sso-identity-federation.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Requirements for SSO identity federation
+type: Decision Record
+status: Draft
+track: IAM
+---
+
+# Introduction
+
+Our assumption is that there are use cases, where CSPs would like to be able to
+let customers access their SCS based services by identifying themselves with
+credentials that are stored and managed external to SCS.
+
+This is based on the observation that prospective customers of a SCS based CSP
+sometimes already come equipped with an IAM solution of their choice, either on
+premises or e.g. as an external 3rd party cloud service. To ease onboarding of
+customer employees (or e.g. customer contracted 3rd party admin staff) as SCS
+users, it would be good to be able to consume these external identities in SCS.
+
+For customers this avoids the neccessity to explicitly maintain an additional
+dedicated account in SCS and this also reduces what SCS needs to do with
+respect to handling storage and lifecycle maintenance for user accounts.
+
+To put it in other workds, in SCS we would like to be able to delegate
+authentication to external identity providers and map those users to roles in
+SCS that can be used for authorization decisions when users access SCS services.
+
+# Motivation for this document
+
+SCS has multiple service layers, like IaaS and CaaS, both of which running their
+own technological stack with specific internal models of accounts and
+authorization. OTOH there are also specific services like the "Status Page"
+application that require consumption of identites.
+
+One thing these services have in common, is that they are able
+to use SSO protocols like OAuth 2.0 or OpenID Connect (OIDC) on top of it to
+delegate authentication. They are service providers (SAML terminology) and can
+be relying parties (OIDC terminology) of a protocol compliant identity provider
+(IdP).
+
+So the idea is, to run an SSO IdP as part of SCS to provide a dedicated point
+of entry for identites, which the SCS service layers can use as a common
+interface to consume external user identities.
+
+The purpose of this document is to sepcify what requirements a specific
+technical IdP implementation (i.e. software solution) needs to fulfill
+in the context of SCS.
+
+# Design Considerations
+
+As a central service for identity handling, the IdP
+service needs to be realy really robust.
+
+Customers shall be able to access self service, so that
+they can make reasonable adjustments e.g. to role mapping.
+At the time of writing this document it's still undecided
+if SCS has the requirement of a decicated "self service"
```

Comment:
```suggestion
if SCS has the requirement of a dedicated "self service" service
```

The latter "service" may be redundant, but also may improve reading flow by differentiating between "self service" (a pattern) and "service" (a thing that can do things like serving a frontend). Maybe there is also better way to express that.

Reply:
I added and removed that additional "service" twice because I was unsure. I accept your proposal.